### PR TITLE
Implementation of Forked Repo version handling

### PR DESCRIPTION
**Pull request by ThilinaManamgoda**
## Automation tests
 - Unit tests 
   > yes
 - Integration tests
   > yes

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (kasunbg) [inline src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java]: Year is 2018! :)

File: src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import org.apache.maven.shared.release.versions.VersionInfo;
+import org.apache.maven.shared.release.versions.VersionParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ForkedRepoVersionInfo extends DefaultVersionInfo {
```
Comment (kasunbg) [inline src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java]: Missing javadocs.

File: src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    private transient Logger log = LoggerFactory.getLogger(ForkedRepoVersionInfo.class);
+
+    public final static Pattern FORKED_REPO_NEXT_DEVELOPMENT_VERSION_PATTERN = Pattern.compile("(.*)(((-wso2v)(([1-9])(\\d*)))(-SNAPSHOT))$");
+    private final static int MAJOR_MINOR_PATCH_VERSION_GROUP_NUM = 1;
```
Comment (kasunbg) [inline src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java]: This constant's purpose is a bit confusing. Shall we improve the constant name?

File: src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    private final static int SNAPSHOT_STR_GROUP_NUM = 8;
```
Comment (kasunbg) [inline src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java]: What is a group number?

File: src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+    private String rootVersion;
+    private Matcher matcher;
+
+    public ForkedRepoVersionInfo(String version) throws VersionParseException {
+        super(version);
+        rootVersion = version;
+        matcher = FORKED_REPO_NEXT_DEVELOPMENT_VERSION_PATTERN.matcher(version);
+        if (!matcher.matches()) {
+            throw new VersionParseException(String.format(Locale.ENGLISH, "Next Development Version (%s) is not a valid version (it must end with \"%s\")",
```
Comment (kasunbg) [inline src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java]: Line length shud be less than 120 characters.

File: src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java (hunk 4 above)
Comment (ThilinaManamgoda) [reply src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java]: Group number is the index number of the respective String within the Pattern

File: src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java (hunk 3 above)
Comment (ThilinaManamgoda) [reply src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java]: how about "DEFAULT_VERSION_GROUP_INDEX

Comment (kasunbg) [reply src/main/java/org/jvnet/hudson/plugins/m2release/ForkedRepoVersionInfo.java]: +1.. `DEFAULT_VERSION_REGEX_GROUP_INDEX`

**Pull request merged**